smart-client-core/src/main/java/com/emc/rest/smart/SmartClientException.java

```diff
@@ -0,0 +1,32 @@
+package com.emc.rest.smart;
+
+public class SmartClientException extends RuntimeException {
+
+    private ErrorType errorType = ErrorType.Unknown;
+
+    public SmartClientException(String message) {
+        super(message);
+    }
+
+    public SmartClientException(String message, Throwable cause) {
+        super(message, cause);
+    }
+    public ErrorType getErrorType() {
+        return errorType;
+    }
+
+    public void setErrorType(ErrorType errorType) {
+        this.errorType = errorType;
+    }
+
+    public enum ErrorType {
+        Client, // 4xx
+        Service, // 5xx
+        Unknown
+    }
+
+    public boolean isServerError() {
```

Comment:
move this above the enum definition (the enum would fall into the category of inner class, which should appear after any methods) - this should be the last change